src/components/sectionListRe.re

```diff
@@ -1,92 +1,153 @@
 external view : ReactRe.reactClass = "SectionList" [@@bs.module "react-native"];
 
-type section 'item =
-  Js.t {
-    .
-    data : array 'item, key : Js.Undefined.t string, renderItem : Js.Undefined.t (renderItem 'item)
-  }
-and renderItem 'item =
-  Js.t {
-    .
-    item : 'item,
-    index : int,
-    section : section 'item,
-    separators :
-      Js.t {
-        .
-        highlight : unit => unit,
-        unhighlight : unit => unit,
-        updateProps : string => Js.t {.} => unit
-      }
-  } =>
-  ReactRe.reactElement
-and separatorProps 'item =
-  Js.t {
-    .
-    highlighted : Js.boolean,
-    leadingItem : Js.Undefined.t 'item,
-    leadingSection : Js.Undefined.t (section 'item),
-    section : section 'item,
-    trailingItem : Js.Undefined.t 'item,
-    trailingSection : Js.Undefined.t (section 'item)
-  }
-and viewToken 'item =
-  Js.t {
-    .
-    item : 'item,
-    key : string,
-    index : Js.undefined int,
-    isViewable : Js.boolean,
-    section : section 'item
+module Make (Item: {type item;}) => {
```

Comment:
`Make` is common in OCaml but I can see your point, so I changed it to `CreateComponent`